packages/yarnpkg-core/sources/StreamReport.ts

```diff
@@ -26,7 +26,7 @@ const GROUP = process.env.GITHUB_ACTIONS
   : process.env.TRAVIS
     ? {start: (what: string) => `travis_fold:start:${what}\n`, end: (what: string) => `travis_fold:end:${what}\n`}
     : process.env.GITLAB_CI
-      ? {start: (what: string) => `section_start:${Date.now() / 1000}:${what}\n`, end: (what: string) => `section_end:${Date.now() / 1000}:${what}\n`}
+      ? {start: (what: string) => `section_start:${Math.floor(Date.now() / 1000)}:${what.toLowerCase().replace(/\W+/g, `_`)}\r\x1b[OK${what}\n`, end: (what: string) => `section_end:${Math.floor(Date.now() / 1000)}:${what.toLowerCase().replace(/\W+/g, `_`)}\r\x1b[OK`}
```

Comment:
```suggestion
      ? {start: (what: string) => `section_start:${Math.floor(Date.now() / 1000)}:${what.toLowerCase().replace(/\W+/g, `_`)}\r\e[OK${what}\n`, end: (what: string) => `section_end:${Math.floor(Date.now() / 1000)}:${what.toLowerCase().replace(/\W+/g, `_`)}\r\e[OK`}
```

According to the [GitLab documentation on custom collapsible sections](https://docs.gitlab.com/ee/ci/pipelines.html#custom-collapsible-sections), the code should be `\r\e[0K` instead of `\r\x1b[OK`. I tested it out and the former works but the latter doesn't.

Reply:
Actually, this doesn't seem to be true. One second while I test it out a bit more.

Reply:
OK I compared the raw logs between a known-good configuration and [a reproduction case](https://gitlab.com/weimeng/ticket-149004/-/jobs/458601439). I think I found the problem, it's a typo!

Let me know if this turns out not to be the cause, more than happy to help troubleshoot further.